src/test/java/github/weichware10/util/db/TrialsTest.java

```diff
@@ -0,0 +1,309 @@
+
+        assertEquals(4, dbClient.trials.getTrialList(null, ToolType.ZOOMMAPS,
+                null,
+                null, 50).size());
+
+        assertEquals(2, dbClient.trials.getTrialList(configIdZm, ToolType.ZOOMMAPS,
+                new DateTime(2021, 12, 12, 11, 0, 0),
+                new DateTime(2021, 12, 12, 13, 30, 0), 50).size());
+
+        assertEquals(1, dbClient.trials.getTrialList(configIdZm, ToolType.ZOOMMAPS,
+                new DateTime(2021, 12, 12, 13, 15, 0),
+                new DateTime(2021, 12, 13, 14, 30, 0), 50).size());
+
+        assertEquals(0, dbClient.trials.getTrialList(configIdZm, ToolType.ZOOMMAPS,
+                new DateTime(2021, 12, 12, 13, 15, 0),
+                new DateTime(2021, 12, 13, 11, 30, 0), 50).size());
+
+        assertEquals(1, dbClient.trials.getTrialList(configIdZm, ToolType.ZOOMMAPS,
+                null,
+                new DateTime(2021, 12, 12, 12, 10, 0), 50).size());
+
+        assertEquals(3, dbClient.trials.getTrialList(configIdZm, ToolType.ZOOMMAPS,
+                new DateTime(2021, 12, 11, 10, 15, 0),
+                null, 50).size());
+
+        assertEquals(3, dbClient.trials.getTrialList(configIdZm, ToolType.ZOOMMAPS,
+                new DateTime(2021, 12, 11, 10, 15, 0),
+                null, 50).size());
+
+        assertEquals(2, dbClient.trials.getTrialList(configIdZm, ToolType.ZOOMMAPS,
+                new DateTime(2021, 12, 11, 10, 15, 0),
+                null, 2).size());
+
+        // CODECHARTS
+        List<String> trialsCc = dbClient.trials.addTrials(configIdCc, 3);
+        TrialData trialData4 = new TrialData(ToolType.CODECHARTS, trialsCc.get(0), configIdCc,
+                new DateTime(2021, 12, 11, 21, 0, 0), "Ja", new ArrayList<DataPoint>());
+        trialData4.addDataPoint(new int[] { 1, 1 }, (new int[] { 1, 1 }));
+        dbClient.trials.setTrial(trialData4);
+
+        TrialData trialData5 = new TrialData(ToolType.CODECHARTS, trialsCc.get(1), configIdCc,
+                new DateTime(2021, 12, 11, 19, 0, 0), "Ja", new ArrayList<DataPoint>());
+        trialData5.addDataPoint(new int[] { 2, 2 }, (new int[] { 2, 2 }));
+        dbClient.trials.setTrial(trialData5);
+
+        TrialData trialData6 = new TrialData(ToolType.CODECHARTS, trialsCc.get(2), configIdCc,
+                new DateTime(2021, 12, 11, 22, 0, 0), "Ja", new ArrayList<DataPoint>());
+        trialData6.addDataPoint(new int[] { 2, 2 }, (new int[] { 2, 2 }));
+        dbClient.trials.setTrial(trialData6);
+
+        String configIdCc2 = dbClient.configurations.setConfiguration(codeConfig);
+        List<String> trialsCc2 = dbClient.trials.addTrials(configIdCc2, 1);
+
+        TrialData trialData8 = new TrialData(ToolType.CODECHARTS, trialsCc2.get(0), configIdCc2,
+                new DateTime(2021, 12, 9, 14, 0, 0), "Ja", new ArrayList<DataPoint>());
+                trialData8.addDataPoint(new int[] { 4, 4 }, new int[] { 4, 4 });
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def' child has incorrect indentation level 16, expected level should be 8.